core/store/src/trie/state_parts.rs

```diff
@@ -152,21 +149,37 @@ impl Trie {
     fn get_trie_nodes_for_part_with_flat_storage(
         &self,
         part_id: PartId,
+        shard_id: ShardId,
```

Comment:
Oh... that should be retrieved from `Trie` because it already knows the shard. But we currently don't have method for retrieving its id. Can we add shard_**uid** method to `FlatStorageChunkView` as it must exist, as a quick&dirty solution?

Reply:
> add shard_uid method to FlatStorageChunkView

Done